packages/sdk/src/cross-chain-provider.ts

```diff
@@ -278,14 +280,110 @@ export class CrossChainProvider implements ICrossChainProvider {
     })
   }
 
+  public async toCrossChainMessage(
+    message: MessageLike
+  ): Promise<CrossChainMessage> {
+    // TODO: Convert these checks into proper type checks.
+    if ((message as CrossChainMessage).message) {
+      return message as CrossChainMessage
+    } else if (
+      (message as TokenBridgeMessage).l1Token &&
+      (message as TokenBridgeMessage).l2Token &&
+      (message as TokenBridgeMessage).transactionHash
+    ) {
+      const messages = await this.getMessagesByTransaction(
+        message as TransactionLike
+      )
+
+      // TODO: Can we do better verification here?
+      const found = messages
+        .sort((a, b) => {
+          return a.logIndex - b.logIndex
+        })
+        .find((m) => {
+          return m.logIndex > (message as TokenBridgeMessage).logIndex
```

Comment:
Should this be `===` ?

Reply:
We need to find the first SentMessage event such that the logIndex is greater than the TokenBridgeMessage logIndex. Basically we detect TokenBridgeMessage events separately (they're usually stuff like `ERC20DepositInitiated` or `WithdrawalInitiated`). These events are then followed by a SentMessage event.

Reply:
I can add a comment here to make this more clear

Reply:
I added a detailed comment that explains what's happening here